nilearn/plotting/tests/test_surf_plotting.py

```diff
@@ -197,9 +197,141 @@ def test_plot_surf_roi_error():
     mesh = generate_surf()
     rng = np.random.RandomState(0)
     roi_idx = rng.randint(0, mesh[0].shape[0], size=5)
-
-    # Wrong input
     with pytest.raises(
             ValueError,
             match='roi_map does not have the same number of vertices'):
         plot_surf_roi(mesh, roi_map=roi_idx)
+
+
+def _generate_img():
+    mni_affine = MNI152TEMPLATE.get_affine()
+    data_positive = np.zeros((7, 7, 3))
+    rng = np.random.RandomState(42)
+    data_rng = rng.rand(7, 7, 3)
+    data_positive[1:-1, 2:-1, 1:] = data_rng[1:-1, 2:-1, 1:]
+    nii = nibabel.Nifti1Image(data_positive, mni_affine)
+    return nii
+
+
+def test_plot_img_on_surf_hemispheres_and_orientations():
+    nii = _generate_img()
+    # Check that all combinations of 1D or 2D hemis and orientations work.
+    plot_img_on_surf(nii, hemispheres='right', views='lateral')
+    plot_img_on_surf(nii, hemispheres=['left', 'right'], views='lateral')
+    plot_img_on_surf(nii,
+                     hemispheres='right',
+                     views=['medial', 'lateral'])
+    plot_img_on_surf(nii,
+                     hemispheres=['left', 'right'],
+                     views=['dorsal', 'medial'])
+
+
+def test_plot_img_on_surf_colorbar():
```

Comment:
also as there seems to be some redundancy in these tests, if we could reduce the number of tests to make it faster that would be great

Reply:
Just came across this. I originally wrote theses tests to ensure that figures would be correctly generated for any combination of rows (hemispheres) and columns (views) in rows: (1, 2+), columns: (1, 2+). This means testing for 1 row and 1 column, 1 row and 2 columns, 2 rows and 1 column and 2 rows and 2 columns.